src/runes/rune.rs

```diff
@@ -0,0 +1,127 @@
+use super::*;
+
+#[derive(Default, Serialize, Debug, PartialEq, Copy, Clone, PartialOrd)]
+pub(crate) struct Rune(pub(crate) u128);
+
+impl Rune {
+  pub(crate) fn minimum_at_height(height: Height) -> Self {
+    Self(u128::from(
+      Sat::SUPPLY - height.starting_sat().0 - height.subsidy(),
+    ))
+  }
+}
+
+impl Display for Rune {
+  fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+    let mut n = self.0;
+    if n == u128::max_value() {
+      return write!(f, "BCGDENLQRQWDSLRUGSNLBTMFIJAV");
+    }
+
+    n += 1;
+    let mut symbol = String::new();
+    while n > 0 {
+      symbol.push(
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+          .chars()
+          .nth(((n - 1) % 26) as usize)
+          .unwrap(),
+      );
+      n = (n - 1) / 26;
+    }
+
+    for c in symbol.chars().rev() {
+      write!(f, "{c}")?;
+    }
+
+    Ok(())
+  }
+}
+
+impl FromStr for Rune {
+  type Err = crate::Error;
+
+  fn from_str(s: &str) -> crate::Result<Self> {
+    let mut x = 0u128;
+    for (i, c) in s.chars().enumerate() {
+      if i > 0 {
+        x += 1;
+      }
+      x *= 26;
+      match c {
+        'A'..='Z' => {
```

Comment:
We need a mode to display the rune in a runic alphabet for the memes, we can use this one easily since it is also 26 characters: https://en.wikipedia.org/wiki/Anglo-Saxon_runes